src/core/server/workspaces/workspace_permission_control.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export enum WorkspacePermissionMode {
+  Read,
+  Admin,
+}
+
+export class WorkspacePermissionControl {
+  public async validate(
+    workspaceId: string,
+    permissionModeOrModes: WorkspacePermissionMode | WorkspacePermissionMode[]
```

Comment:
```suggestion
    permissionModeOrModes: WorkspacePermissionMode | WorkspacePermissionMode[],
    request: OpensearchDashboardRequest,
```
We may need at least the request info for us to know which user is doing the request.

Reply:
Updated